src/e2e-test/java/io/cdap/plugin/gcs/GCSValidationHelper.java

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.gcs;
+
+import au.com.bytecode.opencsv.CSVReader;
+import com.google.api.gax.paging.Page;
+import com.google.cloud.storage.Blob;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageOptions;
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import io.cdap.e2e.utils.PluginPropertyUtils;
+import org.apache.avro.file.DataFileReader;
+import org.apache.avro.file.FileReader;
+import org.apache.avro.generic.GenericDatumReader;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.io.DatumReader;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+/**
+ * GCS Validation Helper.
+ */
+public class GCSValidationHelper {
+  private static final String avroFilePath = PluginPropertyUtils.pluginProp("gcsAvroExpectedFilePath");
+  private static final String csvFilePath = PluginPropertyUtils.pluginProp("gcsCsvExpectedFilePath");
+  private static final String projectId = PluginPropertyUtils.pluginProp("projectId");
```

Comment:
I see that you are using this projectId 2 times but in same method, creating it as a local variable is fine I believe.

Reply:
Using the local variable of projectId inside the function.